### PR TITLE
Add NAT support.

### LINKED ISSUE #246: Doc UPNP
Dev PR: https://github.com/ConsenSys/teku/pull/3490 (check PR merged first)

### PR DESCRIPTION
**Pull request by bgravenorst**
## Pull Request Description
Add instructions to configure NAT via UPnP.

## Fixed Issue(s)
Fixes #246

### CODE REVIEW

File: docs/HowTo/Find-and-Connect/Specifying-NAT.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+Specify `NONE` to explicitly configure the external IP address and ports advertised using
+[`--p2p-host`](../../Reference/CLI/CLI-Syntax.md#p2p-host) and
+[`--p2p-port`](../../Reference/CLI/CLI-Syntax.md#p2p-port) for the P2P service.
+
+!!! important
+
+    When the NAT method is set to `NONE`, the advertised port is the same as the
+    [listening port](../../Reference/CLI/CLI-Syntax.md#p2p-port). Manual configuration of your
+    firewall may be required to allow external hosts to create inbound connections to Teku.
```
Comment (rolfyone) [inline docs/HowTo/Find-and-Connect/Specifying-NAT.md:42]: manual configuration `will` be required

File: docs/HowTo/Find-and-Connect/Specifying-NAT.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+!!! important
+
+    When the NAT method is set to `UPNP`, the advertised port is the same as the
+    [listening port](../../Reference/CLI/CLI-Syntax.md#p2p-port).
+
+## None
+
+Specify `NONE` to explicitly configure the external IP address and ports advertised using
+[`--p2p-host`](../../Reference/CLI/CLI-Syntax.md#p2p-host) and
```
Comment (rolfyone) [inline docs/HowTo/Find-and-Connect/Specifying-NAT.md:35]: So in teku, manual configuration would be set using 
`--p2p-advertised-ip`
`--p2p-advertised-port`
and the advertised port can differ from `--p2p-port`.
For example if you need port 9001 on firewall, but 9000 on teku box, then you'd set advertised port as 9001, and manually configure your firewall to forward 9001 -> 9000 on the teku machine (udp and tcp)
discovery takes care of `--p2p-host`, and teku doesnt offer that configuration option at all.

File: docs/HowTo/Find-and-Connect/Specifying-NAT.md (hunk 1 above)
Comment (rolfyone) [reply docs/HowTo/Find-and-Connect/Specifying-NAT.md:42]: i'd actually probably delete this important note, and just cleanup the definition of none...

File: docs/HowTo/Find-and-Connect/Specifying-NAT.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+!!! tip
+
+    UPnP support is often disabled by default in networking firmware. If disabled by default, you
+    must explicitly enable UPnP support.
+
+!!! important
+
+    When the NAT method is set to `UPNP`, the advertised port is the same as the
+    [listening port](../../Reference/CLI/CLI-Syntax.md#p2p-port).
```
Comment (rolfyone) [inline docs/HowTo/Find-and-Connect/Specifying-NAT.md:30]: this statement is just completely incorrect, i'd remove this note.

The whole point of upnp, is you're telling the firewall to map you an external port to this service, and theres absolutely no guarantee that it'll be the same port you're using to offer the service, in fact it's probably fairly odd to have the same port used.

File: docs/HowTo/Find-and-Connect/Specifying-NAT.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+UPnP in home or small office environments where a wireless router or modem provides NAT isolation.
+
+UPnP automatically detects if a node is running in a UPnP environment and provides port forwarding.
+UPnP might introduce delays during node startup, especially on networks without a UPnP gateway
+device.
+
+!!! tip
+
+    UPnP support is often disabled by default in networking firmware. If disabled by default, you
+    must explicitly enable UPnP support.
```
Comment (rolfyone) [inline docs/HowTo/Find-and-Connect/Specifying-NAT.md:25]: 'if disabled' should say 'it is disabled'

Comment (rolfyone) [reply docs/HowTo/Find-and-Connect/Specifying-NAT.md:25]: actually, i just mis-read, i think this is ok now that i'm re-reading it.

File: docs/HowTo/Find-and-Connect/Specifying-NAT.md (hunk 3 above)
Comment (bgravenorst) [reply docs/HowTo/Find-and-Connect/Specifying-NAT.md:30]: Done.

File: docs/HowTo/Find-and-Connect/Specifying-NAT.md (hunk 2 above)
Comment (bgravenorst) [reply docs/HowTo/Find-and-Connect/Specifying-NAT.md:35]: Thanks, I also updated the `--p2p-advertised-port` description using your example. Hope I understood that correctly.

File: docs/HowTo/Find-and-Connect/Specifying-NAT.md (hunk 1 above)
Comment (bgravenorst) [reply docs/HowTo/Find-and-Connect/Specifying-NAT.md:42]: Done.

File: docs/HowTo/Find-and-Connect/Specifying-NAT.md [hunk 5]
```diff
@@ -24,19 +24,10 @@ device.
...
-
-    When the NAT method is set to `UPNP`, the advertised port is the same as the
-    [listening port](../../Reference/CLI/CLI-Syntax.md#p2p-port).
-
 ## None
 
 Specify `NONE` to explicitly configure the external IP address and ports advertised using
-[`--p2p-host`](../../Reference/CLI/CLI-Syntax.md#p2p-host) and
-[`--p2p-port`](../../Reference/CLI/CLI-Syntax.md#p2p-port) for the P2P service.
-
```
Comment (rolfyone) [inline docs/HowTo/Find-and-Connect/Specifying-NAT.md]: sorry, so p2p-port is still the local listening port, just not the advertised, it's only p2p-host that's not valid for teku

Comment (rolfyone) [reply docs/HowTo/Find-and-Connect/Specifying-NAT.md]: actually i think thats ok

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**